pkg/kubelet/container_bridge.go

```diff
@@ -43,15 +43,16 @@ func createCBR0(wantCIDR *net.IPNet) error {
 		glog.Error(err)
 		return err
 	}
-	// restart docker
+	// Stop docker so that babysitter process can restart it again with proper configurations and
+	// checkpoint file
 	// For now just log the error. The containerRuntime check will catch docker failures.
 	// TODO (dawnchen) figure out what we should do for rkt here.
 	if util.UsingSystemdInitSystem() {
-		if err := exec.Command("systemctl", "restart", "docker").Run(); err != nil {
+		if err := exec.Command("systemctl", "stop", "docker").Run(); err != nil {
```

Comment:
I though ubuntu multi-node didn't, local-up-cluster surely doesn't (though it doesn't use --configure-cbr0), not sure about hyperkube either. I feel like anyone using our salt probably will get a babysitter, but almost no-one uses our salt.

I'm fine if we actually document that having a babysitter is a requirement for a working cluster with --configure-cbr0 (or is that already documented somewhere?). I still feel this is risky. 

> All your suggestion here is we have to duplicate logic in two places: one in babysitter, one in kubelet, which I don't want to.

I'm suggesting `--docker-restart-script=/usr/sbin/docker-checker.sh` and a script isn't specified, fallback to service restart. And a flock in the restart script.